java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java

```diff
@@ -661,13 +669,58 @@ private void createCollection(final eXistMessage em) {
         final Optional<Integer> mode = getMode(metaData);
         final Optional<Long> createTime = getCreationTime(metaData);
 
-        updateOrCreateCollection(sourcePath, userName, groupName, mode, createTime);
+        // Be sure collection exists
+        createOrCheckCollection(sourcePath);
+
+        // Update meta data
+        updateCollectionMetadata(sourcePath, userName, groupName, mode, createTime);
+    }
+
+    /**
+     * Create new collection when required, or force update meta-data when already present.
+     */
+    private Collection createOrCheckCollection(final XmldbURI sourcePath) throws MessageReceiveException {
+
+        // Reference to collection
+        Collection collection = null;
+
+        // New collection to be created
+        try (DBBroker broker = brokerPool.get(Optional.of(securityManager.getSystemSubject()));
+             Txn txn = txnManager.beginTransaction()) {
+
+            setOrigin(txn);
+
+            // Create collection when required
+            collection = broker.getOrCreateCollection(txn, sourcePath);
+
+            if (collection == null) {
+                throw new MessageReceiveException("Collection " + sourcePath.toString() + " does not exist");
+            }
+
+            broker.saveCollection(txn, collection);
+
+            // Commit change
+            txn.commit();
+            broker.flush();
```

Comment:
interesting...... I see this pattern on many places. I expected this is not required, but for sure we need to search [link](https://github.com/eXist-db/exist/search?o=asc&q=%22broker.flush%22+extension%3Ajava&s=indexed&type=Code&utf8=✓) ; thnx for the feedback!